internal/rbd/migration.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ceph/ceph-csi/internal/util"
+	"github.com/ceph/ceph-csi/internal/util/log"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// isMigrationVolHandle check the passed in volHash is a volume handle
+// of migrated volume.
+func isMigrationVolHandle(volHash string) bool {
+	return strings.Contains(volHash, migVersion) && strings.Contains(volHash, migImageNamePrefix)
+}
+
+// parseMigrationVolHandle decode the volume handle and generate a migrationHandle
+// struct which consists of version, mon, image name and pool information.
+func parseMigrationVolHandle(vh string) (*migrationHandle, error) {
+	mh := &migrationHandle{}
+	handSlice := strings.Split(vh, migHandleFieldSep)
+	if len(handSlice) < migHandleTotalLength {
+		// its short of length in this case, so return error
+		return nil, ErrInvalidVolID
+	}
+
+	// Store pool
+	mh.PoolName = strings.Join(handSlice[migHandleSplitLength:], migHandleFieldSep)
+
+	// Parse migration version, mons and image
+	for _, field := range handSlice[:migHandleSplitLength] {
+		switch {
+		case field == migVersion:
+			// version is an identifier of migration handle format tied with a
+			// migration feature version, ex: alpha
+			mh.Version = field
```

Comment:
Do we need this check? what are we doing based on the Version?

Reply:
the version helps us to identify the vol handle. its similar to the version of vol handle we have.